[Go] Can't use WithPromptDir with WithPromptFS

**Describe the bug**
The `WithPromptFS` documentation says you can use `WithPromptDir` to specify the subdirectory within the embedded filesystem, but using them together throws a panic error.

**To Reproduce**
```go
//go:embed my-prompts/*
var promptsFS embed.FS

func main() {
    g := genkit.Init(ctx,
        genkit.WithPromptFS(promptsFS),
        genkit.WithPromptDir("my-prompts"),  // This is suggested in the docs
    )
}
```

Error:
```
panic: genkit.Init: error applying options: cannot use WithPromptDir together with WithPromptFS
```

**Expected behavior**
Based on the docs ([here](https://github.com/firebase/genkit/blob/7444f90a44c1a6476a49ec2faf60f9be201d82aa/go/genkit/genkit.go#L137-L149)), `WithPromptDir` is expected to specify the root directory within the embedded filesystem. Instead it throws an error due to the validation in the `apply` method.

**Runtime (please complete the following information):**
 - OS: macOS
 - Version: 15.7.3

**Go version:**
 -  go1.26.0 darwin/arm64

**Additional context**
The `genkit.Init` function has code that looks like it was meant to support this - it checks `gOpts.PromptDir` when `PromptFS` is set and defaults to "prompts". But the validation prevents `PromptDir` from ever being set alongside `PromptFS`, so that code path is unreachable.

--- a/go/genkit/genkit.go
+++ b/go/genkit/genkit.go
@@ -69,9 +69,6 @@ func (o *genkitOptions) apply(gOpts *genkitOptions) error {
 		if gOpts.PromptDir != "" {
 			return errors.New("cannot set prompt directory more than once (WithPromptDir)")
 		}
-		if gOpts.PromptFS != nil {
-			return errors.New("cannot use WithPromptDir together with WithPromptFS")
-		}
 		gOpts.PromptDir = o.PromptDir
 	}
 
@@ -79,11 +76,7 @@ func (o *genkitOptions) apply(gOpts *genkitOptions) error {
 		if gOpts.PromptFS != nil {
 			return errors.New("cannot set prompt filesystem more than once (WithPromptFS)")
 		}
-		if gOpts.PromptDir != "" {
-			return errors.New("cannot use WithPromptFS together with WithPromptDir")
-		}
 		gOpts.PromptFS = o.PromptFS
-		gOpts.PromptDir = o.PromptDir
 	}
 
 	if len(o.Plugins) > 0 {

--- a/go/genkit/genkit_test.go
+++ b/go/genkit/genkit_test.go
@@ -19,6 +19,7 @@ package genkit
 import (
 	"context"
 	"testing"
+	"testing/fstest"
 
 	"github.com/firebase/genkit/go/core"
 )
@@ -122,4 +123,64 @@ func TestDefineSchemaWithType_Error(t *testing.T) {
 	}
 
 	DefineSchemaFor[Invalid](g)
+}
+
+func TestWithPromptFS(t *testing.T) {
+	tests := []struct {
+		name       string
+		fsys       fstest.MapFS
+		promptDir  string
+		promptName string
+	}{
+		{
+			name: "with custom prompt directory",
+			fsys: fstest.MapFS{
+				"custom-prompts/test.prompt": &fstest.MapFile{
+					Data: []byte(`---
+model: googleai/gemini-2.5-flash
+input:
+  schema:
+    text: string
+---
+{{text}}`),
+				},
+			},
+			promptDir:  "custom-prompts",
+			promptName: "test",
+		},
+		{
+			name: "with default prompts directory",
+			fsys: fstest.MapFS{
+				"prompts/test.prompt": &fstest.MapFile{
+					Data: []byte(`---
+model: googleai/gemini-2.5-flash
+input:
+  schema:
+    text: string
+---
+{{text}}`),
+				},
+			},
+			promptDir:  "", // empty means use default
+			promptName: "test",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			var opts []GenkitOption
+			opts = append(opts, WithPromptFS(tt.fsys))
+			if tt.promptDir != "" {
+				opts = append(opts, WithPromptDir(tt.promptDir))
+			}
+
+			g := Init(ctx, opts...)
+
+			prompt := LookupPrompt(g, tt.promptName)
+			if prompt == nil {
+				t.Fatalf("Expected prompt %q to be loaded", tt.promptName)
+			}
+		})
+	}
 }

